include/zephyr/devicetree/fixed-partitions.h

```diff
@@ -78,15 +78,60 @@ extern "C" {
 #define DT_FIXED_PARTITION_ID(node_id) DT_CAT(node_id, _PARTITION_ID)
 
 /**
- * @brief Get the node identifier of the flash device for a partition
+ * @brief Get the node identifier of the flash memory for a partition
+ * @param node_id node identifier for a fixed-partitions child node
+ * @return the node identifier of the internal memory that contains the
+ * fixed-partitions node, or @ref DT_INVALID_NODE if it doesn't exist.
+ */
+#define DT_MEM_FROM_FIXED_PARTITION(node_id)                                                       \
```

Comment:
I do not like the name, just briefly looking at it I thought it is intended to get some memory mapping, region, or device property  and the description at line 81 got me even more baffled.
Can we have something like `DT_MEM_DEV_NODE_ID_FROM_FIXED_PARTITION` ? Or something.

Reply:
@de-nordic it is consistent with the existing `DT_MTD_FROM_FIXED_PARTITION`